### DISCUSSION TITLE
Calling Action when state property changes

### DISCUSSION THREAD

**Discussion by JokerMartini** [vuejs/pinia#1230] [Help and Questions]
What is the best way for me to trigger an action anytime a specific property has been changed within the state?
In this example the **toggle** property can be mutated from multiple places within the application. Any time **toggle** is mutated, i want to follow it up with executing the action **toggleChanged**. How would this be done? Ideally somehow within the state so the triggering is more easily manageable. 

```javascript
/**
 * General app properties
 */
import { defineStore } from "pinia";
import { collapse } from "myutils";

export const useAppStore = defineStore("app", {
  state: () => ({
    toggle: false,
  }),

  actions: {
    // anytime the toggle property has changed id like to trigger this event
    async toggleChanged(props) {
      console.log('Changed:', this.toggle)
      collapse(this.toggle)
    },
  },
});

```

**Comment by dreambo8563** [comment]
https://pinia.vuejs.org/introduction.html#basic-example
>You can even use a function (similar to a component setup()) to define a Store for more advanced use cases

```ts
export const useToggle = defineStore("toggle", () => {
  const toggle = ref(false);

  const collapse = (v: boolean) => {
    console.info("collapse", v);
  };

  watch(toggle, (t) => {
    console.info("watch toggle", t);
    collapse(t);
  });

  return {
    toggle,
  };
});

```
